Authentication

riencus opened the pull request:
Working and tested authentication and authorization system using spring-security and jwt

riencus committed:
Add Authenticatiocontroller class  and test to handle login request

- AuthenticationController will return a JWT token on a succesfull login
  attempt (The mehtod to generate the token itself hasn't been created
yet)
- Authentication Service is just a dummy class just to pass the test for
  now

new Classess:
- Athens Response and Request
  This class is a base class for all request and response for the API.
This is created cause the UI needs the request and response in a form of
an class, not a primitive object

riencus committed:
Add Credential Class and Repository for authentication purposes

riencus committed:
Add method and test in AuthenticationServiceImpl

- Renaming athensCredential to follow singular convention
- creating method for find by username and password

riencus committed:
Rename Authentication Controller and service to Login to avoid confusion
between spring security authentication and class for generating a new
token

riencus committed:
Change athens credential

Change email & nik to username & password
And include NIK to credential to make sure the credential is owned by someone

riencus committed:
Create (incomplete) authentication mechanism

 Notes:
 - no unit test and not yet tested
 - signUp method is just for manual testing purpose only

 Todo:
 - throw meaningful according to Spring Security specification
 - create unit test

riencus committed:
Create custom AuthenticationEntryPoint and fix Exception

riencus committed:
Half Spring Security implemented

Summary:
- Authentication Process successful
- Authorization not yet implemented

Use Rest-Assured to do integration test of the security configuration

riencus committed:
Fix Authentication Filter

Removing @Component from filter to ensure spring boot don't automatically inject the filter
and several changes to make sure the filter work

riencus committed:
Finished Spring security

Several edit to make the spring security works as expected.
Also change the test from using rest-assured into Spring's mockMvc

riencus committed:
pom.xml: Removing rest-assured dependency

sebastianbagya commented on src/main/java/com/blibli/future/config/WebSecurityConfiguration.java:
```diff
@@ -0,0 +1,58 @@
+        http.csrf().disable()
+                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS).and()
+                .authorizeRequests()
+                .antMatchers("/shift").hasRole("ADMIN")
```
just shift? i think there is more than shift.

riencus replied:
Well, this one is just an example on how to do it, since I'm not too sure which module has which role. 
I hope you guys will help me finish the overall authorization path if possible 😀

riencus committed:
Merge branch 'development' into authentication

riencus committed:
Remove sign up system

Amd rename method on service and repository

riencus committed:
Add Data Seeder

Data seeder class based on Detroit project that will insert dummy data into database at startup

riencus committed:
Merge branch 'development' into authentication

riencus committed:
Authentication system to ignore swagger Api
